Sources/Amplitude/Plugins/iOS/iOSUserInteractionCapture/CustomUIGestureRecognizers.swift

```diff
@@ -0,0 +1,312 @@
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        initialTouchLocation = touches.first?.location(in: captureDelegate.keyWindow)
+        initialRootView = captureDelegate.keyWindow
+        startTime = Date().timeIntervalSince1970
+
+        if let initialTouchLocation,
+           let target = captureDelegate.accessibilityTargets.first(where: {
+            $0.shape.contains(initialTouchLocation) &&
+            ($0.type.contains(.button) || $0.type.contains(.link))
+        }) {
+            initialTarget = target
+        }
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesEnded(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        defer {
+            self.initialTarget = nil
+        }
+
+        guard
+            let initialTouchLocation,
+            let startTime,
+            let initialTarget,
+            let initialRootView,
+            let endTouchLocation = touches.first?.location(in: initialRootView),
+            abs(endTouchLocation.x - initialTouchLocation.x) <= GlobalUIPressGestureRecognizer.pressActivationDelta,
+            abs(endTouchLocation.y - initialTouchLocation.y) <= GlobalUIPressGestureRecognizer.pressActivationDelta
+        else {
+            return
+        }
+
+        let endTime = Date().timeIntervalSince1970
+        let pressDuration = endTime - startTime
+
+        recentTaps.append((initialTarget, endTime))
+        recentTaps = recentTaps.filter { endTime - $0.time <= GlobalUIPressGestureRecognizer.rageClickTimeWindow }
+
+        // A series of continues rage clicks will be eliminated if the frequency of the clicks is higher
+        // than the threshold within the specified interval. This means that when a rage click is detected,
+        // consequent highly frequent clicks on the same element will not be tracked.
+        if recentTaps.count >= GlobalUIPressGestureRecognizer.rageClickCountThreshold {
+            if !isRageClicking {
+                currentRageClickTarget = initialTarget.object
+                isRageClicking = true
+                trackClickRaged()
+
+            } else if initialTarget.object !== currentRageClickTarget {
+                currentRageClickTarget = nil
+                isRageClicking = false
+                recentTaps.removeAll()
+                trackClickNotRaged()
+            }
+        } else {
+            if isRageClicking {
+                currentRageClickTarget = nil
+                isRageClicking = false
+
+            } else {
+                trackClickNotRaged()
+            }
+        }
+
+        func trackClickNotRaged() {
+            var dead = false
+            var longPress = false
+
+            if initialTarget.type.contains(.notEnabled) {
+                dead = true
+            }
```

Comment:
This might provide value for the customers when too many users are clicking on dead buttons (for example a button is not active, but it looks like to be). Fullystory and Dynatrace have this feature for their auto-track.